File: factorial/factorial.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

""" Recursive Python programm to calculate the factorial of a given number.

@File = factorial.py
@Date = 2022-12-28
@Author = Jannis Metrikat
@Version = 1.0

"""

#  Calculates the factorial of a given number.
def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n-1)


# main entry point
n = input("Enter a number to find the factorial of: \n")
print(factorial(int(n)))

File: factorial/factorial.go
// Recursive Go program to calculate the factorial of a given number.
//
// @File = factorial.go
// @Date = 2022-12-28
// @Author = Jannis Metrikat
// @Version = 1.0

package main

import (
	"fmt"
	"os"
	"strconv"
)

// Calculates the factorial of a given number.
func factorial(n int) int {
	if n == 0 {
		return 1
	}
	return n * factorial(n-1)
}

// main is the entry point for the program.
func main() {
	if len(os.Args) != 2 {
		fmt.Println("Usage: go run factorial.go <number>")
		os.Exit(1)
	}

	in := os.Args[1]
	n, err := strconv.Atoi(in)
	if err != nil { panic(err) }

	fmt.Println(factorial(n))
}
